src/ui_basic/multilineeditbox.cc

```diff
@@ -272,14 +274,88 @@ uint32_t MultilineEditbox::Data::snap_to_char(std::string& txt, uint32_t cursor)
 	return cursor;
 }
 
+std::pair<uint32_t, uint32_t> MultilineEditbox::Data::word_boundary(uint32_t cursor,
+                                                                    bool require_non_blank) {
+	uint32_t start = snap_to_char(cursor);
+	uint32_t end = start;
+
+	bool found_non_blank = false;
+	while (start > 0) {
+		uint32_t newpos = prev_char(start);
+		if (isspace(text[newpos]) != 0) {
+			if (!require_non_blank || found_non_blank) {
+				break;
+			}
+		} else {
+			found_non_blank = true;
+		}
+
+		start = newpos;
+	}
+
+	found_non_blank = false;
+	while (end < text.size()) {
+		if (isspace(text[end]) != 0) {
+			if (!require_non_blank || found_non_blank) {
+				break;
+			}
+		} else {
+			found_non_blank = true;
+		}
+
+		end = next_char(end);
+	}
+
+	return {start, end};
+}
+
+std::pair<uint32_t, uint32_t> MultilineEditbox::Data::line_boundary(uint32_t cursor) {
```

Comment:
This is actually more like paragraph boundary, ;) because it only looks for hard `'\n'`-s, but the text in ml. editboxes is also soft wrapped. So triple click often ends up selecting all text.